### ISSUE TITLE
I have a problem with rednderd_apache

### ISSUE THREAD

**Issue by myoldloverock** [chesty/maps-docker-compose#3]
I waited for the maps to load and got an error like. How can I fix it?
`renderd-apache_1      | [Thu Dec 10 12:00:54.609605 2020] [tile:error] [pid 442:tid 140330934429440] [client 172.19.0.13:47972] tile_translate: failed to get valid storage backend, referer: http://192.168.0.10:8000/?z=7&center=-37.985340%2C156.736450&hl=en&alt=0
renderd-apache_1      | ERROR: init_storage_backend: Failed to stat /var/cache/renderd/tiles with error: No such file or directory`

**Comment by chesty** [comment]
I'll double check it works for me from a fresh start. 

In the meantime can you share with me any edits you made to configs and the commands you ran and paste the full logs? maybe double check the logs don't have any passwords or any other private info you don't want public.

My tiles dir appears to be /data/var/lib/mod_tile whereas yours appears to be /var/cache/renderd/tiles, maybe that's the default, so maybe renderd/renderd.conf isn't being mounted correctly into the containers?

**Comment by myoldloverock** [comment]
Im just tuning postgresql, and add diffirent country in `osm.env`.
I recently re-read the logs and saw this error:
```renderd_1             | 2020-12-14 05:04:19+UTC -- /usr/local/bin/renderd-docker-entrypoint.sh:  waiting for renderd-initdb to finish
renderd-initdb_1      | Traceback (most recent call last):
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 169, in _new_conn
renderd-initdb_1      |     conn = connection.create_connection(                                                                                                                                                                                                                                                                                                    renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py", line 96, in create_connection
renderd-initdb_1      |     raise err
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py", line 86, in create_connection
renderd-initdb_1      |     sock.connect(sa)
renderd-initdb_1      | TimeoutError: [Errno 110] Connection timed out
renderd-initdb_1      |
renderd-initdb_1      | During handling of the above exception, another exception occurred:
renderd-initdb_1      |
renderd-initdb_1      | Traceback (most recent call last):
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 699, in urlopen
renderd-initdb_1      |     httplib_response = self._make_request(
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 394, in _make_request
renderd-initdb_1      |     conn.request(method, url, **httplib_request_kw)
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 234, in request
renderd-initdb_1      |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
renderd-initdb_1      |   File "/usr/lib/python3.8/http/client.py", line 1255, in request
renderd-initdb_1      |     self._send_request(method, url, body, headers, encode_chunked)
renderd-initdb_1      |   File "/usr/lib/python3.8/http/client.py", line 1301, in _send_request
renderd-initdb_1      |     self.endheaders(body, encode_chunked=encode_chunked)
renderd-initdb_1      |   File "/usr/lib/python3.8/http/client.py", line 1250, in endheaders
renderd-initdb_1      |     self._send_output(message_body, encode_chunked=encode_chunked)
renderd-initdb_1      |   File "/usr/lib/python3.8/http/client.py", line 1010, in _send_output
renderd-initdb_1      |     self.send(msg)
renderd-initdb_1      |   File "/usr/lib/python3.8/http/client.py", line 950, in send
renderd-initdb_1      |     self.connect()
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 200, in connect
renderd-initdb_1      |     conn = self._new_conn()
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connection.py", line 181, in _new_conn
renderd-initdb_1      |     raise NewConnectionError(
renderd-initdb_1      | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f76cab0b490>: Failed to establish a new connection: [Errno 110] Connection timed out
renderd-initdb_1      |
renderd-initdb_1      | During handling of the above exception, another exception occurred:
renderd-initdb_1      |
renderd-initdb_1      | Traceback (most recent call last):
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 439, in send
renderd-initdb_1      |     resp = conn.urlopen(
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 755, in urlopen
renderd-initdb_1      |     retries = retries.increment(
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py", line 573, in increment
renderd-initdb_1      |     raise MaxRetryError(_pool, url, error or ResponseError(cause))
renderd-initdb_1      | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='www.naturalearthdata.com', port=80): Max retries exceeded with url: /http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f76cab0b490>: Failed to establi
sh a new connection: [Errno 110] Connection timed out'))
renderd-initdb_1      |
renderd-initdb_1      | During handling of the above exception, another exception occurred:
renderd-initdb_1      |
renderd-initdb_1      | Traceback (most recent call last):
renderd-initdb_1      |   File "./scripts/get-external-data.py", line 258, in <module>
renderd-initdb_1      |     main()
renderd-initdb_1      |   File "./scripts/get-external-data.py", line 205, in main
renderd-initdb_1      |     download = s.get(source["url"], headers=headers)
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 555, in get
renderd-initdb_1      |     return self.request('GET', url, **kwargs)
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 542, in request
renderd-initdb_1      |     resp = self.send(prep, **send_kwargs)
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 655, in send
renderd-initdb_1      |     r = adapter.send(request, **kwargs)
renderd-initdb_1      |   File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 516, in send
renderd-initdb_1      |     raise ConnectionError(e, request=request)
renderd-initdb_1      | requests.exceptions.ConnectionError: HTTPConnectionPool(host='www.naturalearthdata.com', port=80): Max retries exceeded with url: /http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f76cab0b490>: Failed to esta
blish a new connection: [Errno 110] Connection timed out'))
renderd-initdb_1      | 2020-12-14 05:04:39+UTC -- /usr/local/bin/renderd-docker-entrypoint.sh: renderd-initdb error downloading shapefiles, exit 11
renderd-initdb_1      | 2020-12-14 05:04:39+UTC -- /usr/local/bin/renderd-docker-entrypoint.sh: deleting shapefiles dir```

**Comment by chesty** [comment]
this url looks weird but it might be ok.

```/http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip```

It looks like the full url is ```http://www.naturalearthdata.com/http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip```

maybe it's just how urllib prints the error.

are you behind a proxy? or maybe a vpn? can you run ```curl http://www.naturalearthdata.com/http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip > /tmp/a.zip``` as a test?

**Comment by myoldloverock** [comment]
Hi! Im check this all is done.I thought that the problem might be in the ubuntu version, the project assembled normally on 18.04 lts, the second server was on 20.04. But it crashed. After rolling back this server to 18.04, the problem remained. I don't know what the problem might be.

**Comment by chesty** [comment]
```
requests.exceptions.ConnectionError: HTTPConnectionPool(host='www.naturalearthdata.com', port=80): Max retries exceeded with url: /http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_boundary_lines_land.zip (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f76cab0b490>: Failed to esta
blish a new connection: [Errno 110] Connection timed out'))
```

this is the problem, well, there might be other problems but this is one that needs to be fixed. Your host doesn't matter.

can you try `docker-compose exec renderd-initdb bash` and run the curl command I gave above and see if it works?

**Comment by myoldloverock** [comment]
Yep! Its works for me. May I write to your email?

**Comment by myoldloverock** [comment]
I compared 2 containers, the first one was created 2 months ago and works correctly and the second one which does not work and writes the error 
```
renderd-apache_1 | ERROR: init_storage_backend: Failed to stat / var / cache / renderd / tiles with error: No such file or directory

renderd-apache_1      | ERROR: init_storage_backend: Failed to stat /var/cache/renderd/tiles with error: No such file or directory
renderd-apache_1      | [Mon Dec 21 21:07:27.547779 2020] [tile:error] [pid 120:tid 139800313067264] [client 172.25.0.11:37728] tile_translate: failed to get valid storage backend, referer: 
```

**In the production container, a search for this line shows** 
```
➜  maps-docker-compose git:(master) ✗ sudo docker exec -it maps-docker-compose_renderd-apache_1 bash
root@05a0846d49c4:/# grep -R '/var/cache/renderd/tiles' /etc
```
**In the second** 
```
novo_tech@maps:~/maps-docker-compose$ sudo docker exec -it maps-docker-compose_renderd-apache_1 bash
root@b31aca63f837:/# grep -R '/var/cache/renderd/tiles' /etc
/etc/apache2/sites-available/renderd.conf:Alias /mod_tile /var/cache/renderd/tiles
/etc/apache2/sites-available/renderd.conf:<Directory /var/cache/renderd/tiles>
/etc/apache2/sites-available/renderd.conf:    ModTileTileDir /var/cache/renderd/tiles
/etc/apache2/sites-enabled/renderd.conf:Alias /mod_tile /var/cache/renderd/tiles
/etc/apache2/sites-enabled/renderd.conf:<Directory /var/cache/renderd/tiles>
/etc/apache2/sites-enabled/renderd.conf:    ModTileTileDir /var/cache/renderd/tiles
root@b31aca63f837:/#
```

what were the changes to the **hammermc/renderd-docker**?
Last changes to hub.docker.com were 17 days ago with latest....

**Comment by chesty** [comment]
upstream changes, I had to change some things around to support them

https://github.com/chesty/renderd-docker/commit/e83a9453a4103e9dc49eeb0f2e3596977cb098fd

So this was a working installation and it stopped working after a pull?

If so, have a look at the apache config files in, /etc/apache2/sites-enabled, it probably has the old files and the new files.
I didn't test upgrading. I tested from a clean slate.
